File: AngularNetCore/ClientApp/src/app/Models/Post.ts
import { User } from "./User";
import { Comment } from "./Comment";

export class Post {
    id: Number;
    userId: Number;
    message: string;
    date: Date;
    expanded: boolean;
    user: User;
    comments: Array<Comment>
    comment: Comment;
    constructor(Message: string = "", UserId: Number = null, id: Number = null) {
        this.id = id;
        this.expanded = false;
        this.userId = UserId;
        this.message = Message;
        this.date = new Date();
        this.comment = new Comment("Teste Comentário");
    }
}

File: AngularNetCore/Models/Post.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace AngularNetCore.Models
{
    public class Post
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int? Id { get; set; }
        public string Message { get; set; }
        public DateTime Date { get; set; }

        public int? UserId { get; set; }
        public User User { get; set; }

        public virtual ICollection<Comment> Comments { get; set; }
    }
}
